modin/core/dataframe/pandas/exchange/dataframe_protocol/dataframe.py

```diff
@@ -0,0 +1,198 @@
+
+    def column_names(self) -> Iterable[str]:
+        for col in self._df.columns:
+            yield col
+
+    def get_column(self, i: int) -> PandasProtocolColumn:
+        return PandasProtocolColumn(
+            self._df.mask(row_positions=None, col_positions=[i]),
+            allow_copy=self._allow_copy,
+            offset=self._offset,
+        )
+
+    def get_column_by_name(self, name: str) -> PandasProtocolColumn:
+        return PandasProtocolColumn(
+            self._df.mask(row_positions=None, col_labels=[name]),
+            allow_copy=self._allow_copy,
+            offset=self._offset,
+        )
+
+    def get_columns(self) -> Iterable[PandasProtocolColumn]:
+        for name in self._df.columns:
+            yield PandasProtocolColumn(
+                self._df.mask(row_positions=None, col_labels=[name]),
+                allow_copy=self._allow_copy,
+                offset=self._offset,
+            )
+
+    def select_columns(self, indices: Sequence[int]) -> "PandasProtocolDataframe":
+        if not isinstance(indices, collections.Sequence):
+            raise ValueError("`indices` is not a sequence")
+
+        return PandasProtocolDataframe(
+            self._df.mask(row_positions=None, col_positions=indices),
+            allow_copy=self._allow_copy,
+            offset=self._offset,
+        )
+
+    def select_columns_by_name(self, names: Sequence[str]) -> "PandasProtocolDataframe":
+        if not isinstance(names, collections.Sequence):
+            raise ValueError("`names` is not a sequence")
+
+        return PandasProtocolDataframe(
+            self._df.mask(row_positions=None, col_labels=names),
+            allow_copy=self._allow_copy,
+            offset=self._offset,
+        )
+
+    def get_chunks(
+        self, n_chunks: Optional[int] = None
+    ) -> Iterable["PandasProtocolDataframe"]:
+        cur_n_chunks = self.num_chunks()
+        n_rows = self.num_rows()
+        offset = 0
+        if n_chunks is None or n_chunks == cur_n_chunks:
+            for length in self._df._row_lengths:
+                yield PandasProtocolDataframe(
+                    self._df.mask(row_positions=range(length), col_positions=None),
+                    allow_copy=self._allow_copy,
+                    offset=offset,
+                )
+                offset += length
+        if n_chunks % cur_n_chunks != 0:
+            raise RuntimeError(
+                "The passed `n_chunks` must be a multiple of `self.num_chunks()`."
+            )
+
+        if n_chunks > n_rows:
+            raise RuntimeError(
+                "The passed `n_chunks` value is bigger than `self.num_rows()`."
+            )
+
+        chunksize = n_rows // n_chunks
+        new_lengths = [chunksize] * n_chunks
+        sum_new_lengths = sum(new_lengths)
+        sum_old_lengths = sum(self._df._row_lengths)
+        if sum_new_lengths < sum_old_lengths:
+            new_lengths[-1] = sum_old_lengths - sum_new_lengths + new_lengths[-1]
```

Comment:
would that work the same?
```suggestion
        new_lengths[-1] = n_rows % chunksize
```

Reply:
No, but this works `new_lengths[-1] = n_rows % n_chunks + new_lengths[-1]`. Changed.